exportlegends.lua

```diff
@@ -346,6 +346,13 @@ function export_more_legends_xml()
                 end
             end
         end
+        if entityV.type == df.historical_entity_type.Guild then -- Get profession
+           for anonK,anonV in pairs(entityV.anon_1) do
```

Comment:
Agreed - anon names are considered unstable (i.e. the numeric suffixes could change at any time) and should not be used in scripts. Have you figured out what this array contains?

Reply:
Turns out that this has since broken - as of https://github.com/DFHack/df-structures/commit/2f93b7188d37be889bb5dc83a758dd13b72d6353, anon_1 through anon_6 point to [these vectors](https://github.com/DFHack/df-structures/blob/2f93b7188d37be889bb5dc83a758dd13b72d6353/df.entities.xml#L662-L667), and `guild_professions` was identified in https://github.com/DFHack/df-structures/commit/1cb6a65f788e0c52674130685a12976f39979905

Reply:
fixed by using new vector 'guild_professions' from df.entities.xml